doc/contributing.md

```diff
@@ -0,0 +1,164 @@
+# How to contribute
+
+* [Discuss your ideas](#discuss-your-ideas)
+* [Clone the repository](#clone-the-repository)
+* [Create your development environment](#create-your-development-environment)
+* [Make your changes](#make-your-changes)
+* [Testing](#testing)
+    * [Run the tests](#run-the-tests)
+    * [Writing your own tests](#writing-your-own-tests)
+* [Open a pull request](#open-a-pull-request)
+* [Optional: Build the documentation](#optional-build-the-documentation)
+* [Optional: Benchmark performance](#optional-benchmark-performance)
+    * [Run the benchmark script](#run-the-benchmark-script)
+    * [Line profiling](#line-profiling)
+    * [Snakeviz visualization](#snakeviz-visualization)
+    * [Memory profiling with memray](#memory-profiling-with-memray)
```

Comment:
Will the internal hyperlinks in this table or contents work correctly in the built documentation? I wasn't entirely sure how to build the docs locally with [pdoc](https://pdoc.dev/), so I haven't tested this.